File: 0011-container-with-most-water/solution.py
class Solution:
	def maxArea(self, height):
		l, r = 0, len(height) - 1
		ans = 0

		while l < r:
			ans = max(ans, (r - l) * min(height[l], height[r]))

			if height[l] <= height[r]:
				l += 1
			else:
				r -= 1 

		return ans

File: 0011-container-with-most-water/solution.java
class Solution {
    public int maxArea(int[] height) {
        int n=height.length;
        int i=0,j=n-1;
        int k=0,l;
        while(i<j){
            l=Math.min(height[i],height[j])*(j-i);
            k=Math.max(l,k);
            if(height[i]>height[j]){
                j--;
            }
            else{
                i++;
            }
        }
        return k;
    }
}
